src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -38,144 +33,99 @@ public LedStrip(int port, int length) {
     }
 
+    private void setSolidColor(Color color) {
+        setSolidColor(color, state.getStartingLed() - 1, state.getEndingLed());
+    }
+
     /**
-     * Updates fadeColor to the correct color for a fade effect
-     * at the current time by interpolating the two given colors.
+     * Updates fadeColor to the correct color for a fade effect at the current time by interpolating
+     * the two given colors.
+     *
      * @param initial Initial color.
-     * @param goal    Final Color.
+     * @param goal Final Color.
      */
     private void updateFade(Color initial, Color goal) {
         Translation3d initialPoint = new Translation3d(initial.red, initial.green, initial.blue);
         Translation3d goalPoint = new Translation3d(goal.red, goal.green, goal.blue);
 
-        double d = initialPoint.getDistance(goalPoint) / fadeDuration;
+        double d = initialPoint.getDistance(goalPoint) / state.getFadeDuration();
         double t = d / (initialPoint.minus(goalPoint)).getNorm();
 
         Translation3d solution = initialPoint.interpolate(goalPoint, t);
-        fadeColor = new Color(
-                (int)solution.getX(),
-                (int)solution.getY(),
-                (int)solution.getZ()
-        );
+        fadeColor = new Color((int) solution.getX(), (int) solution.getY(), (int) solution.getZ());
     }
 
     private void setRainbow() {
         int rainbowHue = 0;
-        for (int i = offset; i < stripLength; i++) {
+        for (int i = state.getStartingLed() - 1; i < state.getEndingLed(); i++) {
             ledBuffer.setHSV(i, rainbowHue, 255, 180);
-            rainbowHue += (180 / stripLength);
+            rainbowHue += (180 / state.getStripLength());
             ledStrip.setData(ledBuffer);
             rainbowHue %= 180;
         }
     }
 
-    @Override
-    public void periodic() {
-        switch (mode){
-            case SOLID:
-                setSolidColor(primary);
+    public Command solid() {
+        return this.run(() -> setSolidColor(state.getPrimary()));
+    }
+
+    public Command percentage(IntSupplier percentage) {
+        return this.run(
+                () -> {
+                    state.setPercentage(percentage.getAsInt());
+                    setSolidColor(
+                            state.getPrimary(),
+                            0,
+                            state.getStripLength() * state.getPercentage() / 100);
+                });
+    }
+
+    public Command percentage() {
+        return percentage(() -> state.getPercentage());
+    }
 
-            case PERCENTAGE:
-                setSolidColor(primary, 0, stripLength*percentage/100);
+    public Command blink(DoubleSupplier blinkTime) {
+        return this.run(
+                () -> {
+                    state.setBlinkTime(blinkTime.getAsDouble());
+                    if (currentColor == state.getPrimary()) currentColor = state.getSecondary();
+                    else currentColor = state.getPrimary();
```

Comment:
You should always wrap blocks with `{ }`. In this case, though, it might be nice to use a ternary condition:
```suggestion
                    currentColor = currentColor == state.getPrimary() ? state.getSecondary() : state.getPrimary();
```
Choose what you think is better.